### ISSUE TITLE
[DEPR]: Remove Enterprise Catalog API V1 from edx-enterprise

### ISSUE THREAD

**Issue by feanil** [openedx/public-engineering#61]
We are in the process of building a new service that will replace the existing enterprise catalog functionality, so we plan to deprecate and remove the existing endpoints at the point when we have the new ones up and running.

We will remove these endpoints specifically:

* EnterpriseCustomerViewSet::contains_content_items (https://github.com/edx/edx-enterprise/blob/master/enterprise/api/v1/views.py#L142)
* All of EnterpriseCustomerCatalogViewSet except for:
** course_detail: https://github.com/edx/edx-enterprise/blob/master/enterprise/api/v1/views.py#L350
** course_run_detail: https://github.com/edx/edx-enterprise/blob/master/enterprise/api/v1/views.py#L378
** program_detail: https://github.com/edx/edx-enterprise/blob/master/enterprise/api/v1/views.py#L406
See here for more details on the migration plan: https://docs.google.com/document/d/1jvmryjFR6KwLdHKbJYpNRwtvW26-gJTBhkPeD2AufEY/edit

### Additional Info
Original Jira Issue: https://openedx.atlassian.net/browse/DEPR-61

**Comment by dianakhuang** [comment]
The replacement service for these v1 apis is: https://github.com/openedx/enterprise-catalog

**Comment by dianakhuang** [comment]
After talking to Enterprise folks here at 2U, we have determined that we have three usages that are still pointing to the old catalog endpoints, but there's only one relevant one that should be implementable on the new enterprise-catalog service.

Usages:
- [ecommerce usage 1](https://github.com/openedx/ecommerce/blob/master/ecommerce/enterprise/utils.py#L152) [ecommerce usage 2](https://github.com/openedx/ecommerce/blob/master/ecommerce/enterprise/utils.py#L515-L560)
- [frontend-app-admin-portal usage](https://github.com/openedx/frontend-app-admin-portal/blob/master/src/data/services/LmsApiService.js#L254-L256)
- [enterprise-data usage](https://github.com/openedx/enterprise-catalog/blob/master/enterprise_catalog/apps/api_client/enterprise.py)


Recommended course of action:
- I do not believe we should remove the EnterpriseCustomer endpoint as part of this work, as even [enterprise-catalog makes calls to it right now](https://github.com/openedx/enterprise-catalog/blob/master/enterprise_catalog/apps/api_client/enterprise.py).
- We should create an endpoint in the enterprise-catalog which returns the data that these usages need or verify that the existing ones cover these usages. We believe the only missing one is that returns catalog ids that match the customer ids.
- We can then replace these usages with corresponding calls to the enterprise-catalog service.
- Then we can remove the enterprise_catalogs endpoint.

**Comment by ruzniaievdm** [comment]
@dianakhuang  I made an additional research and determined that this endpoint already exists in [enterprise-catalogs](https://github.com/openedx/enterprise-catalog/blob/master/enterprise_catalog/apps/api/v1/views/enterprise_catalog_crud.py), some of methods exists in endpoint above. 

As for other methods, like _course_detail_, _course_run_detail_, _program_detail_  the calls these methods in edx-enterprise is no longer used. Instead _ecommerce_, _frontend-app-admin-portal_, _edx-enterprise-data_ repositories called to _enterprise-catalog_ or discovery api.
If this is indeed the case, then I completely remove this endpoint without moving this methods.

**And one more question.**
The response in [ecommerce](https://github.com/openedx/ecommerce/blob/2a5ace5270464637e8e2b9fd54047aed5a7b51f5/ecommerce/enterprise/utils.py#L147) is different from [enterprise-catalog](https://github.com/openedx/enterprise-catalog/blob/1d68b4864168358a3d3ecfd84cba28ebadbe9083/enterprise_catalog/apps/api/v1/views/enterprise_catalog_crud.py#L22). Missed pagination fields like _num_pages_, _current_page_, _start_ .
Does it make sense to add the pagination fields above in the response of _enterpise-catalog_?

**Comment by dianakhuang** [comment]
As long as the usages of the old endpoint get updated, I'm okay with the differing API.

**Comment by robrap** [comment]
@dianakhuang @feanil @ruzniaievdm: Ignore this comment if it is unrelated to this effort, but when I saw V1 catalog, it made me think of the following: https://github.com/edx/api-manager/blob/master/swagger/api.yaml#L31-L37.

**Comment by feanil** [comment]
@robrap @mattdrayer who owns the api-manager? It looks like the v2 endpoints already exist, so we can either just remove the v1 endpoints when they're deleted or we can update them to proxy to the v2 API(I'm not a fan of this since it's a bit confusing).

**Comment by mattdrayer** [comment]
Who is John Galt?  😉

**Comment by feanil** [comment]
>     5. At some point a communication should be sent out to as many external consumers as we can identify, and perhaps a public notice posted somewhere [logical](https://edx-enterprise-api.readthedocs.io/), to inform third parties that the API Gateway V1 endpoints are going away.

@mattdrayer I think now is that point, who can do this and how long a lead time do people need given that the V2 endpoints are already available?

**Comment by mattdrayer** [comment]
Since these are existing enterprise customers it probably makes sense to coordinate with edX/2U Enterprise Customer Support and possibly the edX/2U Enterprise Product Team.

**Comment by mattdrayer** [comment]
I would expect a V1->V2 transition for all enterprise customers/integrators/vendors to take several months, if not longer.

**Comment by feanil** [comment]
@mattdrayer makes sense, thanks for setting expectations. Can you also confirm that the process has started? Or connect me with the right people for starting the process?

**Comment by mattdrayer** [comment]
I've sent you a message in Open edX Slack with some contact info to start with.

**Comment by awais786** [comment]
@feanil https://github.com/openedx/edx-enterprise-data/pull/345 this PR is merged and deployed.

**Comment by robrap** [comment]
[inform] This DEPR is in the Draft state. It can't move forward without a DEPR coordinator to help update it to match the updated DEPR template. No action is required, unless anyone wants to move this out of Draft.

**Comment by dianakhuang** [comment]
Moving to Abandoned until someone decides that they would like to move forward on this.
